src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,68 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  countOfTodos: number;
+  countOfCompletedTodos: number;
+  fetchAddTodo: (title: string) => Promise<boolean>;
+  setErrorMessage: (error: Errors) => void;
+  isInputDisabled: boolean;
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  countOfTodos,
+  countOfCompletedTodos,
+  fetchAddTodo,
+  setErrorMessage,
+  isInputDisabled,
+}) => {
+  const [inputText, setInputText] = useState('');
+  const isAllTodosCompleted = countOfCompletedTodos === countOfTodos;
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (inputText.trim()) {
+      if (await fetchAddTodo(inputText.trim())) {
```

Comment:
The` if (await fetchAddTodo(inputText.trim())) `line could lead to confusion about what happens if the promise rejects. Consider adding a try/catch block around this to handle any potential errors more gracefully.